chord_metadata_service/chord/workflows/wdls/experiments_extended_json.wdl

```diff
@@ -0,0 +1,404 @@
+version 1.0
+
+workflow experiments_extended_json {
+    input {
+        String directory
+        File json_document
+        String drs_url
+        String katsu_url
+        String project_dataset
+        String access_token
+        Boolean validate_ssl
+    }
+
+    call prepare_download_list {
+        input:
+            json_document = json_document,
+            directory = directory
+    }
+
+    ## Calculate the size of the JSON document to determine if there are any URLs to download
+    Int urls_json_size = ceil(size(prepare_download_list.urls_json, "Gi") * 8000)
+
+    if (urls_json_size > 0) {
+        call process_downloads {
+            input:
+                urls_json = prepare_download_list.urls_json
+        }
+
+        call prepare_for_download {
+            input:
+                json_document = prepare_download_list.urls_json
+        }
+
+        scatter (download_info in prepare_for_download.download_list) {
+            call download_file {
+                input:
+                    url = download_info.url,
+                    filename = download_info.filename,
+                    directory = directory,
+                    access_token = access_token,
+                    validate_ssl = validate_ssl
+            }
+        }
+    }
+
+    call prepare_for_drs {
+        input:
+            json_download_list = prepare_download_list.urls_json,
+            json_path_list = prepare_download_list.path_list,
+            downloaded_files = download_file.downloaded_file
+    }
+
+    scatter (path in prepare_for_drs.consolidated_paths_for_drs) {
+        call post_to_drs {
+            input:
+                file_path = path,
+                drs_url = drs_url,
+                project_dataset = project_dataset,
+                token = access_token,
+                validate_ssl = validate_ssl
+        }
+    }
+
+    call write_drs_responses_to_file {
+        input:
+            drs_responses = post_to_drs.response_message
```

Comment:
Not sure if all the scattered `post_to_drs` call responses will be passed here too, or just the last one

Reply:
The `post_to_drs` and `write_drs_responses_to_file`tasks could be combined